parity/main.rs

```diff
@@ -208,6 +208,22 @@ fn latest_exe_path() -> Option<PathBuf> {
 		.and_then(|mut f| { let mut exe = String::new(); f.read_to_string(&mut exe).ok().map(|_| updates_path(&exe)) })
 }
 
+fn set_spec_name_override(spec_name: String) {
+	if let Err(e) = File::create(updates_path("spec_name_overide"))
```

Comment:
> Why is there a need for a file at all? Can't the hypervisor just pass CLI params to a new process?

two reasons:

1. getting data out a child process back to the parent cleanly is non-trivial. i used a return-code thus far but it's no good for information-rich content. i guess we could use std-out but then it wouldn't work properly in all circumstances.

2. this method works well for existing auto-update installations; using the hypervisor to relay information (even if it were easy) would mean everyone manually re-installing before this feature worked, since the hypervisor process is always the most recent _manually_ installed executable. 

> Why not introduce --chain=auto as default setting and just update user_defaults like we do with pruning, tracing and fat_db?

(I actually implemented it this way originally.) `user_defaults` is per-chain, which obv. won't work for the setting of which chain to use. we could use another non-per-chain user-defaults, but then that's a lot of extra baggage for what is really just temporary signalling.